lib/carbonite/migrations/v4.ex

```diff
@@ -0,0 +1,305 @@
+# SPDX-License-Identifier: Apache-2.0
+
+defmodule Carbonite.Migrations.V4 do
+  @moduledoc false
+
+  use Ecto.Migration
+  use Carbonite.Migrations.Version
+  alias Carbonite.Migrations.V1
+
+  @type prefix :: binary()
+
+  @type up_option :: {:carbonite_prefix, prefix()}
+
+  # This buffer ensures that the sequence is initialized to a value hopefully greater than the
+  # current xact id, even if new `transactions` records have been inserted during the migration.
+  @xact_id_buffer 5000
+
+  @impl true
+  @spec up([up_option()]) :: :ok
```

Comment:
`nitpick`

Since the `up_option` type is only used here, maybe it reads better by in-lining it?

```suggestion
  @spec up([{:carbonite_prefix, prefix()}]) :: :ok
```

Reply:
🤷 

copied from the other files, it's `up_option()` everywhere... doesn't really matter anyway, since these files are `@moduledoc false`, so nobody these this besides Dialyzer.